Feature/add as generics

TrN000 opened the pull request:
added generic but tests are broken until constructor is fixed

TrN000 committed:
add gitignore

TrN000 committed:
add testfile for norMmix.

        currently only nor1toMmix under test.

mmaechler commented:
(Ich warte mal ab ..)

TrN000 committed:
misc doc fixes

mmaechler commented:
I don't see in what sense these many changes (notably in `NAMESPACE`)  should be useful. 
Also:
- Don't like to use and depend on `testthat`
- while it's good to have  a  `norMix` |--> to `norMmix` coercion, I'm less convinced about using a generic `as.norMmix`;
  a simpler (and also slightly more efficient) alternative would be to have just 1 function called  e.g.,  `nor1toMmix()`

TrN000 commented:
> I don't see in what sense these many changes (notably in `NAMESPACE`) should be useful. Also:
> 

- NAMESPACE changes were added to appease the autoformatter. I don't particularly care. we can drop it
- most changes to docs were added to pass `check --as-cran`
- made changes to vignette because it was an unfinished document (including notes from me to myself of what still needed to be written).
- change of license because I feel more comfortable with MIT than with a copyleft license.

>     * Don't like to use and depend on `testthat`
>

We have depended on `testthat` since the beginning. All the other tests also run with it.

>     * while it's good to have  a  `norMix` |--> to `norMmix` coercion, I'm less convinced about using a generic `as.norMmix`;
>       a simpler (and also slightly more efficient) alternative would be to have just 1 function called  e.g.,  `nor1toMmix()`

Fair point. Also less of a hassle to export. Can do.

TrN000 commented:
made `as.norMmix` standalone function `nor1toMmix`
rebased a bunch to make structure of work (hopefully) clearer.

check --as-cran passes without notes. will check again against winbuilder and run locally against devel.

mmaechler commented:
> > I don't see in what sense these many changes (notably in `NAMESPACE`) should be useful. Also:
> 
>     * NAMESPACE changes were added to appease the autoformatter. I don't particularly care. we can drop it

Yes, I'd drop them
> 
>     * most changes to docs were added to pass `check --as-cran`

Yes, of course, I did not mean them; sorry for the "too coarse" statement

> 
>     * made changes to vignette because it was an unfinished document (including notes from me to myself of what still needed to be written).
> 
>     * change of license because I feel more comfortable with MIT than with a copyleft license.

really?  Well, I could argue differently.  R  would hardly exist at all if we had not adopted the GPL...

but I can (just barely ;-)  live with the change.
> 
> 
> > ```
> > * Don't like to use and depend on `testthat`
> > ```
> 
> We have depended on `testthat` since the beginning. All the other tests also run with it.
> 

aah... yes, well then it's ok to continue 
(even though:  there is "tinytest" as a viable much less heavy alternative to 'testthat'}  

> > * while it's good to have  a  `norMix` |--> to `norMmix` coercion, I'm less convinced about using a generic `as.norMmix`;
> >   a simpler (and also slightly more efficient) alternative would be to have just 1 function called  e.g.,  `nor1toMmix()`
> > ```
> 
> Fair point. Also less of a hassle to export. Can do.

and you have in the mean time;  thank you!

mmaechler commented on NAMESPACE:
Really, this should only contain the  ` ,  nor1toMmix`  line

mmaechler commented on R/norMmix.R:
I'm pretty sure that  you should just use  `Sig[ ,  , i, drop=FALSE]`   instead of  `matrix(...)` {which is a waste}

TrN000 replied:
this does, as is, not work, because we're working with an array and not a matrix. `isSymmetric` also doesn't work on an array with dim=[1,1,1]. I tried. I might have forgotten the proper way to do it so that it only drops the intended number of dimensions.

mmaechler commented on TODO.md:
remove the       `as.norMmix()` generic + method  ....     and replace with just   `nor1toMmix`

mmaechler commented on DESCRIPTION:
I'm not happy moving away from `GPL`.    If you insist, I can live with `MIT`

TrN000 committed:
add nor1toMmix function

TrN000 committed:
squash: solve p=1 case like this

TrN000 committed:
add GPL3 properly with license text

TrN000 committed:
docfixes: make \code and \link markup instead of markdown

TrN000 committed:
remove Maintainer and author field as already contained in AuthorsatR field

TrN000 committed:
unnecessary link

TrN000 committed:
parcond issue

	unsure wether to include and export or exclude and remove.
	compromise on mentioning as hidden function.

TrN000 committed:
export npar as s3method

	throws --as-cran error otherwise

TrN000 committed:
expand description and add references

	figured out how and where to put citations.
	grepped through repo and added what seemed sensible in the proper places
	no guarantee to completeness

TrN000 committed:
add value sections to man files

TrN000 committed:
bump version patch number

TrN000 committed:
reformat doi links
